crates/query-engine/translation/src/translation/mutation/experimental/delete.rs

```diff
@@ -10,18 +9,21 @@ use query_engine_metadata::metadata;
 use query_engine_metadata::metadata::database;
 use query_engine_sql::sql;
 use std::collections::BTreeMap;
+use tracing;
+
+use super::common;
 
 /// A representation of an auto-generated delete mutation.
 ///
-/// This can get us `DELETE FROM <table> WHERE column = <column_name_arg>`.
+/// This can get us `DELETE FROM <table> WHERE column = <column_name_arg>, ...`.
 #[derive(Debug, Clone)]
 pub enum DeleteMutation {
     DeleteByKey {
         description: String,
         collection_name: String,
         schema_name: sql::ast::SchemaName,
         table_name: sql::ast::TableName,
-        by_column: metadata::database::ColumnInfo,
+        by_columns: Vec<metadata::database::ColumnInfo>,
```

Comment:
nit: would there be any benefit in using nonempty for this?

Reply:
I suppose if we're just `AND` or `OR`-ing expressions together from this, there are obvious `mempty` values, so maybe this is fine as is.

Reply:
Actually yes. I wasn't aware rust had a nonempty vector type, and now I'm looking I don't see one in the stdlib. Not sure which one to choose?

Reply:
Feels pretty similar to the Haskell equivalent.

Reply:
Thanks!